src/backend/distributed/commands/utility_hook.c

```diff
@@ -120,9 +126,12 @@ typedef struct MarkObjectDistributedParams
  * NonMainDbSupportedStatements is an array of statements that are supported
  * from non-main databases.
  */
+ObjectType supportedObjectTypesForGrantStmt[] = { OBJECT_DATABASE };
 static const NonMainDbDistributedStatementInfo NonMainDbSupportedStatements[] = {
-	{ T_GrantRoleStmt, false },
-	{ T_CreateRoleStmt, true }
+	{ T_GrantRoleStmt, false, NULL, 0 },
+	{ T_CreateRoleStmt, true, NULL, 0 },
+	{ T_GrantStmt, false, supportedObjectTypesForGrantStmt,
```

Comment:
where do we make use of third field of those arrays in the code?

Reply:
Disappeared after the merge. I re-added the logic there